src/main/kotlin/eli/inevitable/formatters/TextFormatter.kt

```diff
@@ -0,0 +1,40 @@
+package eli.inevitable.formatters
+
+import eli.inevitable.LINE_BREAK
+import java.io.OutputStream
+import java.nio.charset.Charset
+
+/**
+ * Generalized interface for constructing structured text.
+ */
+interface TextFormatter {
+    /**
+     * @return The total number of lines for the row that includes total padding.
```

Comment:
probably should be called `getRowHeight`, since it feels to me the text formatter will format all rows

Reply:
That could be misleading since I already have a RowFormatter class and a TableFormatter that consumes Rows. So I need the naming to be a bit more general. Though I'm still up for suggestions if you can think of something that fits better.

Reply:
Ah, wasn't aware there were two impls...... maybe remove the words "row" and have "the total number of lines for the component", "the total character width for the component" in the doc, the usage of row seems misleading when it comes to the TableFormatter

Reply:
I adjusted the phrasing as requested.